### DISCUSSION TITLE
What gcode sender programs will this board work with?

### DISCUSSION THREAD

**Discussion by wyrmridr** [phil-barrett/PicoCNC#2] [General]
Hi.  I'm very interested in this board.  Will it work with senders like UGS and OpenBuild's CONTROL?  What others will it work with?

Thanks for any info.

Jay

**Comment by phil-barrett** [comment]
It runs grblHAL so it works with all the standard GCode senders - UGS, OB Control, bCNC, ...  I particularly like ioSender.  Works with LaserGRBL and LightBurn.

In fact, if you know of a GCode Sender that doesn't, please let me know.

**Reply by AlexCH321** [reply]
Hello, I was wondering what it takes to run it on LinuxCNC ? Thank you

**Reply by andrewmarles** [reply]
If you install an ethernet module you can have a look at the RP2040 build of Remora-eth:
https://github.com/scottalford75/Remora-RP2040-W5500

Currently limited to 50 KHz base thread frequency so far less performance than you will get from GRBLHAL (~180 KHz)

**Reply by phil-barrett** [reply]
Does remora rp2040 support shift regs? If not, it won't work on the PicoCNC.

FWIW, 50 KHz is not very good,  grblHAL on the PicoCNC can run at 230 KHz though I only spec it to 200. I think Terje's PIO code is really good.

**Comment by zoro186** [comment]
Hi, I'm building a CNC with the PicoCNC board. I'd like to use Pico CNC with the gSender software. Is this possible? Are there any special precautions to take into account?

Thanks for your attention.

Zorko
